src/components/ingredients/IngredientModal.tsx

```diff
@@ -0,0 +1,331 @@
+import {
+  Button,
+  ButtonGroup,
+  Flex,
+  Heading,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalOverlay,
+  Text,
+} from "@chakra-ui/react";
+import {
+  Dispatch,
+  FC,
+  RefObject,
+  SetStateAction,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import CreateIngredientInput from "./CreateIngredientInput.tsx";
+import ChosenProducts from "./ChosenProducts.tsx";
+import { useQuery } from "@tanstack/react-query";
+import { fetchAll } from "../api/api.ts";
+import { useMyStore } from "../store/store.tsx";
+import { SimpleIngredient, NewIngredient, Product } from "../types.ts";
+import Autocomplete from "./Autocomplete.tsx";
+import IngredientModalTwo from "./IngredientModalTwo.tsx";
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+  focusRef: RefObject<HTMLInputElement>;
+  selectedIngredients: NewIngredient[];
+  setSelectedIngredients: Dispatch<SetStateAction<NewIngredient[]>>;
+};
+
+const IngredientModal: FC<Props> = ({
```

Comment:
Tohle je paradni komponenta. Je trochu komplexnejsi a myslim si, ze je to super priklad, kde by se nam hodil [reducer](https://react.dev/reference/react/useReducer). Kazdopadne netvrdim, ze to musime prepsat. Jen v pripade ze mame tolik stavu, mohlo by nam to zjednodusit zivot o dost.

Protoze treba akce: "setSelectedProduct" by nam rovnou mohla zajistit, ze `error` bude false.
"reset" by zas mohl uvest vsechny ty dilci stavy do defaultniho nastaveni..